### PR TITLE
Dataset

### LINKED ISSUE #10: ProtBertExecutor cannot handle proteins of different lengths.
To reproduce: Change input data file from `samelength.csv` which contains only protein sequences of the same length to a data file with protein sequences of different (e.g. Train_HHblits_1column_short.csv).
Error: 
`pod0@8742[E]:ValueError("Unable to create tensor, you should probably activate truncation and/or padding with 'padding=True' 'truncation=True' to have batched tensors with the same length.")
 add "--quiet-error" to suppress the exception details
Traceback (most recent call last):
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 699, in convert_to_tensors
    tensor = as_tensor(value)
ValueError: expected sequence of length 332 at dim 1 (got 368)`

`During handling of the above exception, another exception occurred:`

`Traceback (most recent call last):
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/jina/peapods/runtimes/zmq/zed.py", line 289, in _msg_callback
    self._zmqlet.send_message(self._callback(msg))
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/jina/peapods/runtimes/zmq/zed.py", line 275, in _callback
    self._pre_hook(msg)._handle(msg)._post_hook(msg)
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/jina/peapods/runtimes/zmq/zed.py", line 221, in _handle
    r_docs = self._executor(
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/jina/executors/__init__.py", line 187, in __call__
    return self.requests[__default_endpoint__](
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/jina/executors/decorators.py", line 103, in arg_wrapper
    return fn(*args, **kwargs)
  File "/Users/georgeamccarthy/Documents/workspace/python/protein_search/protein_search/backend/my_executors.py", line 29, in encode
    encoded_inputs = self.tokenizer(sequences, return_tensors="pt")
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 2310, in __call__
    return self.batch_encode_plus(
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 2495, in batch_encode_plus
    return self._batch_encode_plus(
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/transformers/tokenization_utils.py", line 549, in _batch_encode_plus
    batch_outputs = self._batch_prepare_for_model(
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/transformers/tokenization_utils.py", line 629, in _batch_prepare_for_model
    batch_outputs = BatchEncoding(batch_outputs, tensor_type=return_tensors)
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 204, in __init__
    self.convert_to_tensors(tensor_type=tensor_type, prepend_batch_axis=prepend_batch_axis)
  File "/Users/georgeamccarthy/opt/anaconda3/envs/jina/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 715, in convert_to_tensors
    raise ValueError(
ValueError: Unable to create tensor, you should probably activate truncation and/or padding with 'padding=True' 'truncation=True' to have batched tensors with the same length.`

### PR DESCRIPTION
**Pull request by fissoreg**
🏆 **Enhancements**
- This PR introduces a new dataset and some mechanisms for efficient loading/processing (batching and preprocessing).
- WARNING: the dataset is comprehensive, but needs adjustments (dropping duplicates, mainly).

### Purpose
- Working with a functional prototype.

### Why?
- We need a comprehensive dataset to obtain meaningful search results.

### Feedback required over
- A quick pair of :eyes: on the code
- Discussion on configuration file / directory management

### References (OPTIONAL)
- Refer to #8 #5
- Fix #10

### CODE REVIEW

**Comment by fissoreg** [comment]
@georgeamccarthy this is kind of ready, it could be useful for the fronted integration. It's a draft cause I would like to add some tests and polish the code a bit.

**Comment by fissoreg** [comment]
This might fix #21 if we agree on the directories management.

File: backend/app.py [hunk 1]
```diff
@@ -1,29 +1,41 @@
...
+            fp=data_file,
+            field_resolver={
+                "sequence": "text",
+                "structureId": "id"
+            }
         )
-    )
-    protein = proteins[randint(0, len(proteins)-1)]
-    print(f'Searching with protein sequence:\n{protein.text}')
+        proteins = DocumentArray(docs_generator)[0:42]
```
Comment (georgeamccarthy) [inline backend/app.py:26]: What getting 42? Is this arbitrary to limit the computation time in development.

Or maybe it has something to do with li[fe, the Universe, and everything.](https://en.wikipedia.org/wiki/42_(number)#The_Hitchhiker's_Guide_to_the_Galaxy)

Comment (fissoreg) [reply backend/app.py:26]: It's definitely both ;)

File: frontend/app.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# %%
```
Comment (georgeamccarthy) [inline frontend/app.py:1]: Are we gonna have compatibility issues with the [frontend branch?](https://github.com/georgeamccarthy/protein_search/tree/fronted)

**Review by georgeamccarthy — COMMENTED** [review]
Looks good!

File: frontend/app.py (hunk 2 above)
Comment (fissoreg) [reply frontend/app.py:1]: Well, compatibility shouldn't be a problem but some care will be needed during the merge. It shouldn't be hard, anyways.

**Comment by georgeamccarthy** [comment]
Could you please put a link somewhere e.g. in the README of where this dataset is from? I couldn't find its source by following the download link.

**Comment by fissoreg** [comment]
Sure, we should update the README. For now the link is in #8

**Comment by fissoreg** [comment]
Tests are still missing but I'm gonna merge this one to work on the frontend. I'll open an issue for the tests.

File: backend/utils.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# %%
+import os.path
+
+from urllib.request import urlretrieve
+
+def load_or_download(url, filename):
+  if not os.path.isfile(filename):
+    print(f"{filename} not found, downloading from {url}")
+    r = urlretrieve(url, filename)
```
Comment (Rubix982) [inline backend/utils.py:9]: This could use a DocString.

**Comment by Rubix982** [comment]
The PR looks okay. I didn't understand somethings since I'm unfamiliar with `jina`.

Just one thing, @fissoreg , do you think this PR is too 'large'? It touches 10 files together over 8 commits (excluding merge from main). Maybe @georgeamccarthy can better give an idea of this, since again, I'm unfamiliar with the code base.

File: tests/test_executors.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 from my_executors import *
```
Comment (Rubix982) [inline tests/test_executors.py:8]: Preferably, `*` shouldn't be used. It's confusing and ambiguous.

One way would be to write this as,

```python
import my_executors as executors
executors.my_class()
executors.my_func()
```

**Review by Rubix982 — COMMENTED** [review]
Just a minor aspect for code readability purposes.

File: tests/test_executors.py [hunk 5]
```diff
@@ -2,13 +2,13 @@
 import sys
 
 sys.path.append("../")
+sys.path.append("./backend/")
 
 # %%
-from protein_search.backend.my_executors import *
-from jina import DocumentArray, Flow
-from jina.types.document.generators import from_csv
+from my_executors import *
```
Comment (Rubix982) [inline tests/test_executors.py:8]: This can be changed to,

```python
import my_executors as executors
executors.my_class()
executors.my_func() 
```

**Review by Rubix982 — COMMENTED** [review]
Changes requestd

**Comment by georgeamccarthy** [comment]
> The PR looks okay. I didn't understand somethings since I'm unfamiliar with `jina`.
> 
> Just one thing, @fissoreg , do you think this PR is too 'large'? It touches 10 files together over 8 commits (excluding merge from main). Maybe @georgeamccarthy can better give an idea of this, since again, I'm unfamiliar with the code base.
> 
> The commit messages can be improved, 'Frontend layout' seems unsure, 'Added frontend layouts' seems better. Also no periods at the end of the commit messages.
> 
> And another commit, `Missing file.` was confusing as it only adds 3 lines and I didn't really understand the connection of the commit message with the code written.
> 
> PS: Sorry for the 2 reviews below. GitHub is acting buggy.

We haven't agreed on a naming convention for commits, PRs and the size of each of these. We could do that moving forward. I agree that the commit messages could be clearer as well as potentially splitting this into two PRs, leaving the frontend to another or change the title to e.g. Add new dataset and basic frontend.

**Pull request merged**